### PR TITLE
Integrate Sigma Rules parser for importing rules based on Sigma schema format-part 1

### PR DESCRIPTION
**Pull request by sbcd90**
### Description
integrate sigma rules-types, exeptions, utils

## Design
image (1)

 
### Issues Resolved
[Integrate Sigma Rules parser for importing rules based on Sigma schema format](https://github.com/opensearch-project/security-analytics/issues/5)
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/rules/types/Placeholder.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.securityanalytics.rules.types;
+
+public class Placeholder {
```
Comment (lezzago) [inline src/main/java/org/opensearch/securityanalytics/rules/types/Placeholder.java:7]: Can you write a TODO comment what this is a placeholder for?

Comment (sbcd90) [reply src/main/java/org/opensearch/securityanalytics/rules/types/Placeholder.java:9]: @lezzago added TODO comment

**Review by getsaurabh02 — APPROVED** [review]
Thanks for the changes, LGTM. Let's open a primary issue for Sigma Parser and add the related/planned part PRs to it for the full context. We can then resolve the issue once all related PRs are out.

**Comment by sbcd90** [comment]
yes @getsaurabh02 i created the issue https://github.com/opensearch-project/security-analytics/issues/5

**Pull request merged**